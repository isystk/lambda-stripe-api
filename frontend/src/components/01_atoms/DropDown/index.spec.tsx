import React from 'react'
import { render } from '@testing-library/react'
import DropDown, { DropDownProps } from './index'

import { useRouter } from 'next/router'

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}))


describe('LandingPageTemplate', () => {
  it('Match Snapshot', () => {
    useRouter.mockImplementationOnce(() => ({
      route: '/',
      pathname: '/',
      query: {},
      asPath: '/',
    }))
    const props: DropDownProps = {
      label: 'サンプル太郎',
      items: [
        { label: 'リンク1', link: 'http://link1' },
        { label: 'リンク2', link: 'http://link2' },
        { label: 'リンク3', link: 'http://link3' },
      ],
    }
    const { asFragment } = render(<DropDown {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
