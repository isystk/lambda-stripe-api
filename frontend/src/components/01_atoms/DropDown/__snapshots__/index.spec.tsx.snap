// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPageTemplate Match Snapshot 1`] = `
<DocumentFragment>
  <button
    class="css-0 flex items-center justify-between w-full py-2 pl-3 pr-4 text-gray-900 rounded hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-blue-700 md:p-0 md:w-auto dark:text-white md:dark:hover:text-blue-500 dark:focus:text-white dark:border-gray-700 dark:hover:bg-gray-700 md:dark:hover:bg-transparent"
    data-dropdown-toggle="dropdownNavbar"
    id="dropdownNavbarLink"
  >
    サンプル太郎
    <svg
      aria-hidden="true"
      class="w-5 h-5 ml-1"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
  <div
    class="hidden z-10 font-normal bg-white divide-y divide-gray-100 rounded-lg shadow w-44 dark:bg-gray-700 dark:divide-gray-600"
    id="dropdownNavbar"
  >
    <ul
      aria-labelledby="dropdownLargeButton"
      class="py-2 text-sm text-gray-700 dark:text-gray-400"
    >
      <li>
        <a
          class="block px-4 py-2 hover:bg-gray-100 dark:hover:bg-gray-600 dark:hover:text-white"
          href="#"
        >
          リンク1
        </a>
      </li>
      <li>
        <a
          class="block px-4 py-2 hover:bg-gray-100 dark:hover:bg-gray-600 dark:hover:text-white"
          href="#"
        >
          リンク2
        </a>
      </li>
      <li>
        <a
          class="block px-4 py-2 hover:bg-gray-100 dark:hover:bg-gray-600 dark:hover:text-white"
          href="#"
        >
          リンク3
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
